include/swift/ABI/Executor.h

```diff
@@ -174,10 +175,98 @@ class ExecutorRef {
     return Implementation & WitnessTableMask;
   }
 
-  bool operator==(ExecutorRef other) const {
+  bool operator==(SerialExecutorRef other) const {
     return Identity == other.Identity;
   }
-  bool operator!=(ExecutorRef other) const {
+  bool operator!=(SerialExecutorRef other) const {
+    return !(*this == other);
+  }
+};
+
+/// Deprecated name for "SerialExecutorRef"; When it was first introduced it was ExecutorRef,
+/// but it always meant specifically a serial executor.
+typedef SerialExecutorRef ExecutorRef;
```

Comment:
Do we need this?  Is there code using this header outside our repository?

Reply:
Probably not, let me remove this.

@rokhinip I don't think you're using the actual header from Swift anywhere right? If anything it would have been a copy I expect.